File: 42-trapping-rain-water/42-trapping-rain-water.java
class Solution {
    public int trap(int[] height) {
        int [] arr1 = new int[height.length];
        int [] arr2 = new int[height.length];
        int max = 0;
        for ( int i=0;i<height.length; i++){
            arr1[i]=Math.max(height[i],max);
            max=Math.max(height[i],max);
        }
        max=0;
        for ( int i=height.length-1;i>-1; i--){
            arr2[i]=Math.max(height[i],max);
            max=Math.max(height[i],max);
        }
        int water=0;
        for(int i=0;i<height.length;i++){
            water+=Math.min(arr1[i],arr2[i])-height[i];
        }
        return water;
    }
}

File: 42-trapping-rain-water/42-trapping-rain-water.py
class Solution:
    def trap(self, height: List[int]) -> int:
        l=[]
        r=[0]*len(height)
        
        c=0
        
        for i in height:
            c=max(c,i)
            l.append(c)
        c=0
        for i in range(len(height)-1,-1,-1):
            c=max(c,height[i])
            r[i]=c
            
        c=0
        for i in range(len(height)):
            c+=min(l[i],r[i])-height[i]
        
        return c